From 605c3d8356f430475f2a3086b9bf5b6397e495c5 Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@freescale.com>
Date: Tue, 21 Jul 2015 09:09:10 +0530
Subject: [PATCH 0505/1383] mc-bus: Do not destroy DPMCP MC I/O object on
 device remove

We do not create MC I/O object in device add, so we should not
destroy the MC I/O object on device remove. MC I/O object for
DPMCP is created in fsl_mc_portal_allocate() and freed in
fsl_mc_portal_free().

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@freescale.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/staging/fsl-mc/bus/mc-bus.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/fsl-mc/bus/mc-bus.c b/drivers/staging/fsl-mc/bus/mc-bus.c
index 4521720..85e59a6 100644
--- a/drivers/staging/fsl-mc/bus/mc-bus.c
+++ b/drivers/staging/fsl-mc/bus/mc-bus.c
@@ -598,12 +598,8 @@ void fsl_mc_device_remove(struct fsl_mc_device *mc_dev)
 
 		if (&mc_dev->dev == fsl_mc_bus_type.dev_root)
 			fsl_mc_bus_type.dev_root = NULL;
-	} else if (strcmp(mc_dev->obj_desc.type, "dpmcp") == 0) {
-		if (mc_dev->mc_io) {
-			fsl_destroy_mc_io(mc_dev->mc_io);
-			mc_dev->mc_io = NULL;
-		}
-	}
+	} else
+		WARN_ON(mc_dev->mc_io != NULL);
 
 	kfree(mc_dev->driver_override);
 	mc_dev->driver_override = NULL;
-- 
2.8.1

